src/vm/method.cpp

```diff
@@ -4850,6 +4850,8 @@ BOOL MethodDesc::SetStableEntryPointInterlocked(PCODE addr)
 
     PCODE pExpected = GetTemporaryEntryPoint();
     PTR_PCODE pSlot = GetAddrOfSlot();
+    // how can we get here if the vtvable slots aren't writeable?
```

Comment:
I don't think they're not writable... They're just almost never written to (). Also - this code runs for normal JITed methods as well, where the MT is obviously writable... 
I'm surprised this only fires sometimes - I would expect this assert to fail pretty much always.
If you look into MD::DoBackpatch - it can (and will) write to pretty much any MT slot.
It doesn't go through this method to do so.. but still. DoPrestub can get here for pretty much any method as well (assuming somebody tries to call the method through a temporary entry point).